atomic_reactor/util.py

```diff
@@ -1279,3 +1280,30 @@ def get_log_files(self, osbs, build_id):
 class DefaultKeyDict(dict):
     def __missing__(self, key):
         return key
+
+
+def get_platforms_in_limits(workflow, input_platforms=None):
+    def make_list(value):
+        if not isinstance(value, list):
+            value = [value]
+        return value
+
+    if not input_platforms:
+        return None
+    excluded_platforms = set()
+    only_platforms = set()
+    if not isinstance(input_platforms, set):
+        expected_platforms = set(input_platforms)
+    else:
+        expected_platforms = deepcopy(input_platforms)
+    build_file_dir = workflow.source.get_build_file_path()[1]
+    container_path = os.path.join(build_file_dir, REPO_CONTAINER_CONFIG)
+    if os.path.exists(container_path):
+        with open(container_path) as f:
+            data = yaml.safe_load(f)
+            if data and 'platforms' in data and data['platforms']:
+                excluded_platforms = set(make_list(data['platforms'].get('not', [])))
+                only_platforms = set(make_list(data['platforms'].get('only', [])))
+                if only_platforms:
+                    expected_platforms = expected_platforms & only_platforms
+    return expected_platforms - excluded_platforms
```

Comment:
maybe we could write warning, in case when in container.yaml is same platform in 'only' and 'not' ?
I am aware that it wasn't changed in this PR.

Reply:
Fixed.